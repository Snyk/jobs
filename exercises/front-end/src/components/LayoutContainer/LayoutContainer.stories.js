import '~/assets/scss/styles.scss';
import LayoutContainer from './LayoutContainer';

export default {
  title: 'Library/Base Components/Layout/Container',
  component: LayoutContainer,
};

export const Default = () => ({
  components: { LayoutContainer },
  template: `
    <LayoutContainer>
      <p>Applies max-width and centering of content within.</p>
       
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent ipsum nunc,
      vehicula sed placerat non, condimentum vitae neque. Cras facilisis libero
      justo, eu iaculis massa condimentum ac. Integer sagittis nulla eget varius
      fermentum. Morbi cursus feugiat ex, id condimentum purus fringilla ut.
      Suspendisse sed condimentum massa. Duis interdum volutpat consequat. Aenean ut
      nibh augue. Donec ut lorem nisl.</p>
      
      <p>Etiam vitae purus felis. Nunc ultricies, nunc et accumsan tristique, justo odio 
      fermentum purus, eget aliquet dui leo ut quam. Maecenas eu velit eget lorem sodales 
      aliquam. Sed lorem nibh, imperdiet posuere tincidunt quis, porta at urna. Sed turpis 
      velit, vulputate vel elit id, tincidunt blandit risus. Vestibulum purus magna, 
      tempor posuere luctus eu, pharetra quis tellus. Aliquam viverra eget mauris at 
      vehicula. Vestibulum pretium lacus ut pretium viverra. In id nibh sagittis, iaculis 
      ex quis, aliquet elit. Quisque leo elit, auctor vel gravida nec, laoreet ut est. Duis
      eleifend lobortis congue. Donec feugiat neque nisi, at scelerisque nunc
      dignissim et.</p>
    </LayoutContainer>
  `,
});
