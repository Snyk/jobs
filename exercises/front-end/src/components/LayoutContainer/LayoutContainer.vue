<template>
  <div class="vue--layout-container">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'LayoutContainer',
  status: 'ready',
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/utils';

.vue--layout-container {
  max-width: global(site, maxWidth);
  margin-left: auto;
  margin-right: auto;
  padding-left: space(m) - 4px; // Adjustment to match legacy styles.
  padding-right: space(m) - 4px; // Adjustment to match legacy styles.
}
</style>
