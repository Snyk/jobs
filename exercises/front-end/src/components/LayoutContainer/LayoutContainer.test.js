import { shallowMount } from '@vue/test-utils';
import LayoutContainer from '~/components/LayoutContainer/LayoutContainer';

describe('LayoutContainer', () => {
  it('renders a container', () => {
    const wrapper = shallowMount(LayoutContainer);

    expect(wrapper).toMatchSnapshot();
  });
});
