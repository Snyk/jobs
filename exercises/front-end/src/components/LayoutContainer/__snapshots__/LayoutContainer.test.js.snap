// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutContainer renders a container 1`] = `<div class="vue--layout-container"></div>`;
